### ISSUE TITLE
Simpler Closed Captions (subtitle) selection

### ISSUE THREAD

**Issue by gordonfreeman01** [yuliskov/SmartTubeNext#1908]
To make the closed captions (subtitle) selection experience much better please consider adding:
1. Removing the multiple menus one must go through to select sub language and make sub language the first option to show up when clicking on the button with a fast on/off option accessible in the list
2. Remember subtitle selection between videos instead of having to re-enable per-video
3. Add support for the TEXT button on remotes (as with YT app)

Subtitles are an important and useful accessibility and general usability feature and it would be great for this fantastic app to have better support for them.

**Comment by PeterNjeim** [comment]
The behaviour has changed in the latest update. Clicking the subtitle button will enable or disable the subtitles. Long pressing will open the menu you are already familiar with

**Comment by va1entin** [comment]
> The behaviour has changed in the latest update. Clicking the subtitle button will enable or disable the subtitles. Long pressing will open the menu you are already familiar with

I have just updated to 16.37 but the short click for subtitle disabling doesn't work. It still opens the side menu and interestingly all possible languages are being displayed as subtitle languages regardless of whether there's actually subtitles available for each of them since this update.

Loving SmartTube to bits and was actually waiting for quicker subtitle deactivation so if this would work more reliably in a future version or someone could point me at what I might be doing wrong I'd be grateful!

**Comment by PeterNjeim** [comment]
There are more languages because of the inclusion of auto-translated versions. I haven't looked at the new code so I don't know which subtitle track it's translating from though.

Quick enabling and disabling works for me.

**Comment by kinng** [comment]
I think the YT web based caption interface is the most efficient.  What's missing here is the automatic selection of the default caption.  E.g, a video has included only one caption, then that caption should be used, instead of the translated caption that's been previously set. I.e, there is a need to separate translated caption from video included caption.  To put it into logic:

If caption is short clicked to enable
      If there is an included caption from the video
             Display included caption
      Else
             If translatable, 
                  If there was one translation selected
                       Display translated caption
                  Else
                       Select translation caption

**Comment by PeterNjeim** [comment]
I'm not sure you understand how this works. The captions are always translated by Google, every video has it. Every subtitle track, whether auto-generated or not, is already translated to all of YouTube's supported languages.

If there is only one subtitle track, then it will be translated to all the various languages, and SmartTubeNext will select your preferred language, as it should (or else how would you read them).

**Comment by kinng** [comment]
You are correct in that I don't know how behind the scene captions work.  But as a user, I prefer seeing the video's originally included caption. E.g., if I am watching a Chinese language video AND it has a Chinese language caption, I want the Chinese captions displayed instead of English which is what I want translated to when I am viewing other (e.g. Russian) language videos that are translatable.  This is the current capability of the web based YT feature that I get on the Brave Browser.  For example, on the Brave Browser's YT page and viewing this Chinese language video https://youtu.be/c1Mkh_tftrc
The CC button can be turned on or off. When on, the caption displayed is either Chinese, or an Auto Translated Language, all were set by the settings button.   And most importantly, I did not set any preference via the settings button and Chinese was displayed automatically when I enable CC for I think that's the only included caption.  Now if I view another English language video, like https://youtu.be/2aTULNg5F1E
and enable captions, I will see English displayed.  

This is the feature I wish that's available in this app.

**Comment by PeterNjeim** [comment]
I understand now, and it's an easy feature to add, though I wouldn't recommend it as I think YouTube's method of never automatically choosing an auto-translated subtitle ruins the experience. A better method would be to choose a list of languages you understand, and if one of those languages is present in the subtitle track list, to choose that one, and if not, to auto-translate from another language to your most preferred language.

**Comment by kinng** [comment]
Yes, that will work.

**Comment by garret** [comment]
To me the best way would be to choose a priority from a list of languages. Like English, Italian, Danish, French. I don't want to scroll to the infinite list of all the web translated languages like Chinese or Russian which for sure I will never use since I don't speak them. The app should prioritize captions included in the video and only if these are not available, then switch to the translated ones.

**Comment by PeterNjeim** [comment]
I was also thinking about that, scrolling through the whole list is indeed cumbersome. A language list (which is ordered by priority), chooses the subtitle that is appears highest in the list, and if none are present, to auto-translate from the existing subtitle track to your most preferred language.

And secondly to add a settings option to disable non-preferred languages from appearing in the auto-translated list (and maybe even a step further, remove human-written subtitles that aren't in your preferred language list).

**Comment by jack5432109** [comment]
cc @yuliskov

**Comment by PeterNjeim** [comment]
The new update separates regular subtitles from auto generated ones, though I still do not prefer this. I would rather the app look at my language list (the one in Android Settings) and only show those subtitles (with a checkbox in the SmartTubeNext settings).

**Comment by kinng** [comment]
I think there are at least 2 main use cases for subtitles: to translate foreign languages, and to aid the hearing-impaired.  But we also have 2 kinds of users, Bilinguals and Monolinguals.  For the Monolinguals, the 2 use cases can be implemented simply as an on/off switch for a previously chosen language.  The original single menu option and the current separation of “Subtitles” and “Subtitles (Auto)” options worked fine, in that the chosen language will be remembered and used when the subtitle switch is turn on.  For the Bilinguals and the hearing-impaired use case, the current separation of options offered slightly improved way of choosing the included subtitle.  But we have to choose each time a different language video is viewed. The Google YouTube app implements perfectly for Bilinguals and the hearing-impaired use case, but suffers from the auto translation use case.  I think the challenge here is to be able to switch to the desired language with just the subtitle on/off switch. 

As for logic, I think we have to have one desired Auto translation language, be that chosen from a list or default to the system language.  We also need to have a subtitle mode: always translate, or always use included subtitle first.  The latter mode will always display the first included subtitle and if none included, do the auto translate.

**Comment by fu8765** [comment]
A couple more items to look into.

1) auto generated subtitles of live video. I don't know how this works in the youtube app, but it will be a nice to have to automatically generate subtitles for live streaming.

2) Auto generated and translated subs don't always work. even with clear audio, there are sections of the video that don't get generated CCs, again, not sure how this works on the original app.
